app/views/articles/show.html.erb

```diff
@@ -95,7 +95,9 @@
       <% if !@article.published %>
         <div class="crayons-notice crayons-notice--danger mb-4">
           <strong>Unpublished Post.</strong> This URL is public but secret, so share at your own discretion.
-          <a href="<%= @article.path %>/edit" class="fw-bold">Click to edit</a>.
+          <% if user_signed_in? && policy(@article.object).update? %>
```

Comment:
I call object method because @article is an instance of ArticleDecorator. If I pass the ArticleDecorator class object to policy method, Pundit infers ArticleDecoratorPolicy class, not ArticlePolicy class.